PERF: Series.dropna with non-nan dtype blocks

sinhrks opened the pull request:
Minor improvement when `dropna` actually does nothing. If OK for 0.17, I'll add a release note.

```
import numpy as np
import pandas as pd

s = pd.Series(np.random.randint(0, 1000, 50000000))

# before
%timeit s.dropna()
#1 loops, best of 3: 1.58 s per loop

# after
%timeit s.dropna()
#1 loops, best of 3: 568 ms per loop
```

sinhrks committed:
PERF: Series.dropna with non-nan dtypes

jorisvandenbossche commented on asv_bench/benchmarks/series_methods.py:
```diff
@@ -71,4 +71,24 @@ def setup(self):
+    def setup(self):
+        self.s = Series(np.random.randint(1, 10, 1000000))
+
+    def series_dropna_int64(self):
```
the function that actually get times should start with 'time'

kawochen replied:
Also this `dropna` only has effect for each time `setup` is run, which is not every iteration
And set seed (not sure)?

jreback commented:
sure, pls add a note in 0.17.1 whatsnew

jreback commented:
@sinhrks looks good, can you rebase / add a release note

sinhrks commented:
Rebased, and added release note. Followings are asv result added in this PR.

```
    before     after       ratio
  [a89b96d4] [b2cb56a8]
    10.76ms    10.60ms      0.99  series_methods.series_dropna_datetime.time_series_dropna_datetime
-   12.07ms   726.61μs      0.06  series_methods.series_dropna_int64.time_series_dropna_int64
```

jreback commented:
thanks!
